### ISSUE TITLE
Actor crashed: RuntimeError: can't add a new key into hash during iteration

### ISSUE THREAD

**Issue by brianvans** [kostya/eye#162]
Running into this crash fairly regularly with eye 0.8.  I'm running unicorn under eye using a config that looks pretty similar to the unicorn example.

```
10.02.2016 17:48:34 ERROR -- [celluloid] Actor crashed!
RuntimeError: can't add a new key into hash during iteration
        /usr/lib64/ruby/gems/2.0.0/gems/eye-0.8/lib/eye/process/children.rb:35:in `[]='
        /usr/lib64/ruby/gems/2.0.0/gems/eye-0.8/lib/eye/process/children.rb:35:in `block in add_or_update_children'
        /usr/lib64/ruby/gems/2.0.0/gems/eye-0.8/lib/eye/process/children.rb:33:in `each'
        /usr/lib64/ruby/gems/2.0.0/gems/eye-0.8/lib/eye/process/children.rb:33:in `add_or_update_children'
        /usr/lib64/ruby/gems/2.0.0/gems/eye-0.8/lib/eye/process/watchers.rb:23:in `block in add_watchers'
        /usr/lib64/ruby/gems/2.0.0/gems/eye-0.8/lib/eye/process/watchers.rb:48:in `call'
        /usr/lib64/ruby/gems/2.0.0/gems/eye-0.8/lib/eye/process/watchers.rb:48:in `block in add_watcher'
        /usr/lib64/ruby/gems/2.0.0/gems/celluloid-0.17.3/lib/celluloid/actor.rb:339:in `block in task'
        /usr/lib64/ruby/gems/2.0.0/gems/celluloid-0.17.3/lib/celluloid/task.rb:44:in `block in initialize'
        /usr/lib64/ruby/gems/2.0.0/gems/celluloid-0.17.3/lib/celluloid/task/fibered.rb:14:in `block in create'
```

**Comment by kostya** [comment]
this is quite strange, can you 100% reproduce it? it yes try also gem eye `v0.8.celluloid15` (which is used old celluloid version)

this error happens when modifying hash in iteration:
```
ruby -e 'x = {:a => :b}; x.each { |k, v| x["asdf"] = 1 } '
-e:1:in `block in <main>': can't add a new key into hash during iteration (RuntimeError)
```

but there is nothing similar in the eye code.

**Comment by brianvans** [comment]
I've yet to figure out a way to reliably reproduce this.  As far as configured checks for this process group, there is a memory check and an http health check for the master and memory checks for the child processes.

**Comment by brianvans** [comment]
Here's another stack trace of when this condition was encountered during a restart, which might be helpful:

    10.02.2016 00:58:30 ERROR -- [celluloid] Actor crashed!
    RuntimeError: can't add a new key into hash during iteration
            /usr/lib64/ruby/gems/2.0.0/gems/eye-0.8/lib/eye/process/children.rb:35:in `[]='
            /usr/lib64/ruby/gems/2.0.0/gems/eye-0.8/lib/eye/process/children.rb:35:in `block in add_or_update_children'
            /usr/lib64/ruby/gems/2.0.0/gems/eye-0.8/lib/eye/process/children.rb:33:in `each'
            /usr/lib64/ruby/gems/2.0.0/gems/eye-0.8/lib/eye/process/children.rb:33:in `add_or_update_children'
            /usr/lib64/ruby/gems/2.0.0/gems/eye-0.8/lib/eye/process/children.rb:4:in `add_children'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/eval_helpers.rb:58:in `evaluate_method'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/callback.rb:191:in `block in run_methods'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/callback.rb:190:in `each'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/callback.rb:190:in `run_methods'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/callback.rb:159:in `call'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/transition.rb:450:in `block (2 levels) in after'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/transition.rb:450:in `each'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/transition.rb:450:in `block in after'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/transition.rb:448:in `catch'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/transition.rb:448:in `after'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/transition.rb:251:in `run_callbacks'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/transition_collection.rb:127:in `run_callbacks'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/transition_collection.rb:63:in `block (2 levels) in perform'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/transition_collection.rb:63:in `catch'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/transition_collection.rb:63:in `block in perform'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/transition_collection.rb:182:in `block in within_transaction'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/transition.rb:227:in `block in within_transaction'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/machine.rb:1902:in `block in within_transaction'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/machine.rb:2222:in `transaction'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/machine.rb:1902:in `within_transaction'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/transition.rb:226:in `within_transaction'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/transition_collection.rb:181:in `within_transaction'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/transition_collection.rb:62:in `perform'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/transition.rb:219:in `perform'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/event.rb:172:in `fire'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/event.rb:247:in `block in add_actions'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/machine.rb:765:in `call'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/machine.rb:765:in `block (2 levels) in define_helper'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/event.rb:252:in `block in add_actions'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/machine.rb:765:in `call'
            /usr/lib64/ruby/gems/2.0.0/gems/state_machine-1.2.0/lib/state_machine/machine.rb:765:in `block (2 levels) in define_helper'
            /usr/lib64/ruby/gems/2.0.0/gems/eye-0.8/lib/eye/process/states.rb:11:in `switch'
            /usr/lib64/ruby/gems/2.0.0/gems/eye-0.8/lib/eye/process/commands.rb:78:in `restart_process'
            /usr/lib64/ruby/gems/2.0.0/gems/eye-0.8/lib/eye/process/controller.rb:23:in `restart'
            /usr/lib64/ruby/gems/2.0.0/gems/eye-0.8/lib/eye/process/scheduler.rb:49:in `scheduled_action'
            /usr/lib64/ruby/gems/2.0.0/gems/celluloid-0.17.3/lib/celluloid/calls.rb:28:in `public_send'
            /usr/lib64/ruby/gems/2.0.0/gems/celluloid-0.17.3/lib/celluloid/calls.rb:28:in `dispatch'
            /usr/lib64/ruby/gems/2.0.0/gems/celluloid-0.17.3/lib/celluloid/call/sync.rb:16:in `dispatch'
            /usr/lib64/ruby/gems/2.0.0/gems/celluloid-0.17.3/lib/celluloid/cell.rb:50:in `block in dispatch'
            /usr/lib64/ruby/gems/2.0.0/gems/celluloid-0.17.3/lib/celluloid/cell.rb:76:in `block in task'
            /usr/lib64/ruby/gems/2.0.0/gems/celluloid-0.17.3/lib/celluloid/actor.rb:339:in `block in task'
            /usr/lib64/ruby/gems/2.0.0/gems/celluloid-0.17.3/lib/celluloid/task.rb:44:in `block in initialize'
            /usr/lib64/ruby/gems/2.0.0/gems/celluloid-0.17.3/lib/celluloid/task/fibered.rb:14:in `block in create'

**Comment by kostya** [comment]
fixed in 0.8.1
